python/lsst/ts/ess/labjack/base_labjack_data_client.py

```diff
@@ -0,0 +1,177 @@
+# This file is part of ts_ess_labjack.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["BaseLabJackDataClient"]
+
+import asyncio
+import concurrent
+import logging
+import socket
+import types
+from collections.abc import Callable
+from typing import Any
+
+# Hide mypy error `Module "labjack" has no attribute "ljm"`
+from labjack import ljm  # type: ignore
+
+from lsst.ts.ess import common
+
+from lsst.ts import salobj
+
+# Time limit for connecting to the LabJack (seconds)
+CONNECT_TIMEOUT = 5
+
+# LabJack's special identifier to run in simulation mode.
+MOCK_IDENTIFIER = "LJM_DEMO_MODE"
+
+
+class BaseLabJackDataClient(common.BaseDataClient):
+    """Base class for ESS data clients that read a LabJack T7 or similar.
+
+    Parameters
+    ----------
+    name : str
```

Comment:
Please use back ticks around types in this and the next line.

Reply:
Good suggestion. This is copied from BaseDataClient in ts_ess_common so we'll want to fix it there at some point. I have updated my local copy but would rather have additional changes before making a PR.